src/components/forms/form-sign-up.jsx

```diff
@@ -1,60 +1,95 @@
 import React from 'react';
 import { Form } from 'reactstrap';
-
+import PropTypes from 'prop-types';
 import { FormElement } from './form-element';
 
-const elementsOfFormSignUp = [
-  {
-    id: 0,
-    name: 'name',
-    label: 'Ваше имя',
-    placeholder: 'имя',
-    type: 'text',
-  },
-  {
-    id: 1,
-    name: 'number',
-    label: 'Номер телефона',
-    placeholder: 'number',
-    type: 'number',
-  },
-  {
-    id: 2,
-    name: 'signUpEmail',
-    label: 'Email',
-    placeholder: 'email',
-    type: 'email',
-  },
-  {
-    id: 3,
-    name: 'signUpPassword',
-    label: 'Пароль',
-    placeholder: 'password',
-    type: 'password',
-  },
-  {
-    id: 4,
-    name: 'signUpPassword',
-    label: 'Повторите пароль',
-    placeholder: 'password',
-    type: 'password',
-  },
-];
+export const FormSignUp = ({
+  errors, onSubmit, register, control, watch,
+}) => {
+  const makeElementsOfFormSignUp = () => [
+    {
+      id: 0,
+      name: 'name',
+      label: 'Ваше имя',
+      placeholder: 'имя',
+      type: 'text',
+      rulesValidation: { required: true },
+      errorMessage: 'Please enter name',
+    },
+    {
+      id: 1,
+      name: 'number',
+      label: 'Номер телефона',
+      placeholder: 'number',
+      type: 'number',
+      rulesValidation: { required: true, minLength: 8 },
+      errorMessage: 'Please enter number',
+    },
+    {
+      id: 2,
+      name: 'signUpEmail',
+      label: 'Email',
+      placeholder: 'email',
+      type: 'email',
+      rulesValidation: { required: true, pattern: /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/ },
+      errorMessage: 'Please enter correct email',
+    },
+    {
+      id: 3,
+      name: 'signUpPassword',
+      label: 'Пароль',
+      placeholder: 'password',
+      type: 'password',
+      rulesValidation: { required: true, pattern: /^(?=.*[a-z])(?=.*[A-Z])(?=.*[0-9])(?=.{8,})/ },
+      errorMessage: 'Please enter password that contain at least 1 lowercase, 1 uppercase, 1 numeric, minimum  8 symbols',
+    },
+    {
+      id: 4,
+      name: 'signUpPasswordRepeat',
+      label: 'Повторите пароль',
+      placeholder: 'password',
```

Comment:
repeat password. Try to keep the same sense  for label and placeholder, otherwise it will be confusing